parachain/primitives/core/src/outbound.rs

```diff
@@ -192,6 +206,15 @@ impl Command {
 			),
 		}
 	}
+
+	/// The priority of the command
+	pub fn priority(&self) -> Priority {
+		match self {
+			Command::Upgrade { .. } => Priority::High,
+			Command::SetOperatingMode { .. } => Priority::High,
+			_ => Priority::Normal,
+		}
+	}
 }
```

Comment:
The priority of the command.

Reply:
I think we should determine the priority using the `AggregateMessageOrigin` origin of the command, because its most efficient.

The implementation of `ProcessMessage::process_message` needs to decode each message to determine the priority. Suppose there are 10000 low priority messages in the queue, followed by 1 high priority message. After delivering 100 low priority messages for processing, the maximum block weight may be reached, and processing stops. And then the situation repeats in the next block.

Ultimately what happens is that the high priority message is never processed because its at the end of the queue.

Instead, since `ProcessMessage::process_message` is also passed an origin parameter, we should use that. Specifically, we should consider the `AggregateMessageOrigin::Parachain(BridgeHub)` origin to be high priority.

Reply:
https://github.com/Snowfork/snowbridge/pull/949/commits/239f3c37a261c7bfda76fb6dcfbce5805eeecee4 for the fix.

> And then the situation repeats in the next block. Ultimately what happens is that the high priority message is never processed because its at the end of the queue.

From my previous unit test this is not true, seems the `pallet_message_queue` process message from different queues in a round-robin manner, so after the first 100 low-priority messages get processed in the first block the high-priority message from another parachain will get the chance in the next block.

But anyway, the fix above as your suggestion is better, will get the 1 high-priority message and 99 low-priority messages in the first block, which is exactly what we want.  Unit test related covering in https://github.com/Snowfork/snowbridge/blob/239f3c37a261c7bfda76fb6dcfbce5805eeecee4/parachain/pallets/outbound-queue/src/test.rs#L315

Reply:
Though I removed `priority` from the submit interface in the fix above,  still some concerns about always deeming message from bridge hub as high priority, specifically assuming message queue for bridge hub has already been spammed by the `create_agent` message, the `upgrade` will never get the chance to execute.

But consider the fee problem has been addressed in https://github.com/Snowfork/snowbridge/pull/968, maybe we can make it simple in this PR and try to improve it later.